server/npg/porch/endpoints/tasks.py

```diff
@@ -27,133 +28,159 @@
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {"description": "Unexpected error"}
+    }
 )
 
+
 @router.get(
     "/",
     response_model=List[Task],
     summary="Returns all tasks.",
-    description="Return all tasks. A filter will be applied if used in the query."
+    description='''
+    Return all tasks. A filter will be applied if used in the query.
+    The filter feature is not yet implemented.'''
 )
-async def get_tasks(db_accessor=Depends(get_DbAccessor)):
+async def get_tasks(
+    db_accessor=Depends(get_DbAccessor),
+    permission=Depends(validate)
+) -> List[Task]:
+
     return await db_accessor.get_tasks()
 
-#@router.get(
-#    "/{task_name}",
-#    response_model=Task,
-#    summary="Get one task.",
-#    description="Get one task using its unique name."
-#)
-#def get_task(task_name: str):
-#    return Task(name=task_name)
 
 @router.post(
     "/",
     response_model=Task,
-    summary="Creates one task.",
     status_code=status.HTTP_201_CREATED,
     responses={
         status.HTTP_201_CREATED: {"description": "Task creation was successful"},
-        status.HTTP_404_NOT_FOUND: {"description": "The pipeline for this task is invalid"},
         status.HTTP_409_CONFLICT: {"description": "A task with the same signature already exists"}
-    }
-)
-async def create_task(task: Task, db_accessor=Depends(get_DbAccessor)):
-    """
+    },
+    summary="Creates one task record.",
+    description='''
     Given a Task object, creates a database record for it and returns
-    the same object with status 201 'Created'
+    the same object, the response HTTP status is 201 'Created'. The
+    new task is assigned pending status, ie becomes awailable for claiming.
 
     The pipeline specified by the `pipeline` attribute of the Task object
-    should exist. If it does not exist, return status 404 'Not found' and
-    an error.
+    should exist. If it does not exist, return status 404 'Not found'.'''
+)
+async def create_task(
+    task: Task,
+    db_accessor=Depends(get_DbAccessor),
+    permission=Depends(validate)
+) -> Task:
 
-    Errors if task status is not PENDING.
-    """
+    _validate_request(permission, task.pipeline)
     created_task = None
     try:
-        created_task = await db_accessor.create_task(token_id=1, task=task)
+        created_task = await db_accessor.create_task(
+            token_id=permission.requestor_id,
+            task=task
+        )
     except IntegrityError:
         raise HTTPException(
             status_code=409,
             detail='Unable to create task, as another like it already exists'
         )
     except NoResultFound:
         raise HTTPException(status_code=404, detail='Failed to find pipeline for this task')
+
     return created_task
 
+
 @router.put(
     "/",
     response_model=Task,
-    summary="Update one task.",
     responses={
         status.HTTP_200_OK: {"description": "Task was modified"},
-        status.HTTP_404_NOT_FOUND: {
```

Comment:
Should we be removing the openAPI docs that this would display? Or do we consider 404 to be self-explanatory?

Reply:
I had to remove tests, which were expecting 404 for not found pipelines, explained this in a comment in tests. 404 for task not found is retained.